be/src/runtime/client_cache.h

```diff
@@ -108,6 +112,9 @@ class ClientCacheHelper {
     // MetricRegistry
     bool _metrics_enabled;
 
+    // max connections in this cache, -1 means unlimited
+    int _max_cache_size;
```

Comment:
I think you have a max cache size for all connection is bad. this should used for each endpoint. If one cluster has 100 backends. there is at least one connection for each, which is 100 as total. 

So, I think you should rethink about this, and reimplement it.

Reply:
I change it to size per host